jupyter_client/jsonutil.py

```diff
@@ -106,7 +108,21 @@ def json_default(obj):
     if isinstance(obj, datetime):
         obj = _ensure_tzinfo(obj)
         return obj.isoformat().replace('+00:00', 'Z')
-    elif isinstance(obj, bytes):
+
+    if isinstance(obj, bytes):
         return b2a_base64(obj).decode('ascii')
-    else:
-        raise TypeError("%r is not JSON serializable" % obj)
+
+    if (
+        isinstance(obj, (set, types.GeneratorType))
+        or hasattr(obj, '__iter__')
+        and hasattr(obj, '__next__')
+    ):
```

Comment:
Why not (with `from collections.abc import Iterable`):
```suggestion
    if isinstance(obj, Iterable):
```

Reply:
Thanks!